app/main.py

```diff
@@ -25,190 +1,257 @@
-            mordred["hp"] += mordred["potion"]["effect"]["hp"]
+            if "hp" in mordred["potion"]["effect"]:
+                mordred["hp"] += mordred["potion"]["effect"]["hp"]
 
-    # red_knight
-    red_knight = knightsConfig["red_knight"]
+        # red_knight
+        red_knight = knightsConfig["red_knight"]
 
-    # apply armour
-    red_knight["protection"] = 0
-    for a in red_knight["armour"]:
-        red_knight["protection"] += a["protection"]
+        # apply armour
+        red_knight["protection"] = 0
+        for a in red_knight["armour"]:
+            red_knight["protection"] += a["protection"]
 
-    # apply weapon
-    red_knight["power"] += red_knight["weapon"]["power"]
+        # apply weapon
+        red_knight["power"] += red_knight["weapon"]["power"]
 
-    # apply potion if exist
-    if red_knight["potion"] is not None:
-        if "power" in red_knight["potion"]["effect"]:
-            red_knight["power"] += red_knight["potion"]["effect"]["power"]
+        # apply potion if exist
+        if red_knight["potion"] is not None:
+            if "power" in red_knight["potion"]["effect"]:
+                red_knight["power"] += red_knight["potion"]["effect"]["power"]
 
-        if "protection" in red_knight["potion"]["effect"]:
-            red_knight["protection"] += red_knight["potion"]["effect"]["protection"]
+            if "protection" in red_knight["potion"]["effect"]:
+                red_knight["protection"] += red_knight["potion"]["effect"]["protection"]
 
-        if "hp" in red_knight["potion"]["effect"]:
-            red_knight["hp"] += red_knight["potion"]["effect"]["hp"]
+            if "hp" in red_knight["potion"]["effect"]:
+                red_knight["hp"] += red_knight["potion"]["effect"]["hp"]
 
-    # -------------------------------------------------------------------------------
-    # BATTLE:
+        # -------------------------------------------------------------------------------
+        # BATTLE:
 
-    # 1 Lancelot vs Mordred:
-    lancelot["hp"] -= mordred["power"] - lancelot["protection"]
-    mordred["hp"] -= lancelot["power"] - mordred["protection"]
+        # 1 Lancelot vs Mordred:
+        lancelot["hp"] -= mordred["power"] - lancelot["protection"]
+        mordred["hp"] -= lancelot["power"] - mordred["protection"]
 
-    # check if someone fell in battle
-    if lancelot["hp"] <= 0:
-        lancelot["hp"] = 0
+        # check if someone fell in battle
+        if lancelot["hp"] <= 0:
+            lancelot["hp"] = 0
 
-    if mordred["hp"] <= 0:
-        mordred["hp"] = 0
+        if mordred["hp"] <= 0:
+            mordred["hp"] = 0
 
-    # 2 Arthur vs Red Knight:
-    arthur["hp"] -= red_knight["power"] - arthur["protection"]
-    red_knight["hp"] -= arthur["power"] - red_knight["protection"]
+        # 2 Arthur vs Red Knight:
+        arthur["hp"] -= red_knight["power"] - arthur["protection"]
+        red_knight["hp"] -= arthur["power"] - red_knight["protection"]
```

Comment:
Similar to the previous comment, the damage calculation here should use the `receive_damage` method of the `Knight` class to ensure consistency and avoid negative damage values.